flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/storage/TieredStorageProducerClient.java

```diff
@@ -119,7 +145,66 @@ private void writeAccumulatedBuffers(
     private void writeAccumulatedBuffer(
             TieredStorageSubpartitionId subpartitionId, Buffer accumulatedBuffer)
             throws IOException {
-        // TODO, Try to write the accumulated buffer to the appropriate tier. After the tier is
-        // decided, then write the accumulated buffer to the tier.
+        int subpartitionIndex = subpartitionId.getSubpartitionId();
+        int tierIndex = chooseStorageTierIfNeeded(subpartitionId);
+
+        Buffer compressedBuffer = compressBufferIfPossible(accumulatedBuffer);
+        updateStatistics(compressedBuffer);
+        boolean isLastBufferInSegment =
+                tierProducerAgents.get(tierIndex).write(subpartitionId, compressedBuffer);
```

Comment:
`TierProducerAgent#write` lacks docs for the return value semantics. I thought the return-value mean whether the buffer is successfully written. How does the agent know whether it's the last buffer without knowing the size of next buffer?

Reply:
OK, I have changed the logic of choosing a new tier. 

The return value of `write` show whether the record can be written to the tier. If true, continue writing. If not, it will choose a new storage tier.